frontend/src/store/store-types.ts

```diff
@@ -30,8 +30,39 @@ export type Task = {
   readonly complete: boolean;
   readonly inFocus: boolean; // Whether the task is in focus
   readonly children: Set<string>;
+  readonly type: 'ONE_TIME' | null; // Null for legacy
 };
 
+/**
+ * The task type without id and every field optional.
+ * Fields that are filled represent differences from master template
+ */
+
+export type PartialTask = Partial<Pick<Task, MainTaskProperties | 'children' | 'type' >>;
+
+export type RepeatMetaData = {
+  readonly startDate: Date;
+  readonly endDate: Date;
+  readonly pattern: RepeatingPattern;
+}
+
+export type ForkedTask = PartialTask & {
```

Comment:
Probably name it `ForkedTaskMetadata` because it does not contain all the info of the forked task